DifferenceGenerator/src/main/kotlin/DifferenceGenerator.kt

```diff
@@ -132,18 +208,28 @@ class DifferenceGenerator(
         image2: BufferedImage,
     ): Frame {
         val differences = getColoredBufferedImage(Color.BLACK)
+        val differencesData = (differences.raster.dataBuffer as DataBufferByte)
 
-        // using a BufferedImage.raster.dataBuffer or just .raster might be faster
-        for (x in 0 until width) {
-            for (y in 0 until height) {
-                val frame1Pixel = image1.getRGB(x, y)
-                val frame2Pixel = image2.getRGB(x, y)
-                if (frame1Pixel - frame2Pixel != 0) {
-                    differences.setRGB(x, y, Color.RED.rgb)
-                }
+        val data1 = (image1.raster.dataBuffer as DataBufferByte).data
+        val data2 = (image2.raster.dataBuffer as DataBufferByte).data
+        var index = 0
+
+        while (index < this.height * this.width * 3) {
+            val blue1 = data1[index] and 0xFF.toByte()
+            val green1 = data1[index + 1] and 0xFF.toByte()
+            val red1 = data1[index + 2] and 0xFF.toByte()
```

Comment:
My IDE recommend extracting this duplicated code into a separate functions but I don't really think that it's already necessary.

Reply:
outdated